routers/repo/migrate.go

```diff
@@ -188,3 +191,28 @@ func MigratePost(ctx *context.Context, form auth.MigrateRepoForm) {
 
 	handleMigrateError(ctx, ctxUser, err, "MigratePost", tpl, &form)
 }
+
+// CancelMigration cancels a running migration
+func CancelMigration(ctx *context.Context) {
+	if !ctx.Repo.IsOwner() {
+		ctx.Error(http.StatusNotFound)
+		return
+	}
+
+	if ctx.Repo.Repository.Status != models.RepositoryBeingMigrated {
+		ctx.Error(http.StatusConflict, "repo already migrated")
+		return
+	}
+
+	// TODO: get task from queues and kill it
+	// https://github.com/go-gitea/gitea/pull/12917/files#r492109921
+
+	if err := repo_service.DeleteRepository(ctx.User, ctx.Repo.Repository); err != nil {
+		ctx.ServerError("DeleteRepository", err)
+		return
+	}
+	log.Trace("Repository deleted: %s/%s", ctx.Repo.Owner.Name, ctx.Repo.Repository.Name)
+
+	ctx.Flash.Success(ctx.Tr("repo.settings.deletion_success"))
```

Comment:
Maybe add a "The migration has been cancelled" here instead.

Reply:
@silverwind done